bin/install.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env python
+
+import math
+import os
+import shutil
+import sys
+import tempfile
+from subprocess import call
+
+
+_is_win = sys.platform == 'win32'
+
+if not _is_win:
+    sys.stdin = open('/dev/tty', 'r')
+
+if sys.version_info[0] == 2:
+    input = raw_input
+
+prompt = os.environ.get('LEKTOR_SILENT') is None
+
+def get_confirmation():
+    if prompt is False:
+        return
+
+    while True:
+        user_input = input('Continue? [Yn] ').lower().strip()
+
+        if user_input in ('', 'y'):
+            print('')
+            return
+
+        if user_input == 'n':
+            print('')
+            print('Aborted!')
+            sys.exit()
+
+def fail(message):
+    print('Error: %s' % message)
+    sys.exit(1)
+
+def multiprint(*lines):
+    for line in lines:
+        print(line)
+
+
+class Installer(object):
```

Comment:
I see what you are going for here, and I assume this is also what you're not completely satisfied with?

To me it looks simpler to declare a separate `install_posix` and `install_windows`. I get that we risk that they diverge but they should be a lot easier to troubleshoot. The bits that are used by both should probably be moved to separate functions.

Reply:
The thing that bothers me in this solution is that the methods manage their own state anyway. `compute_location` will assign to different things depending on in which class it resides, and subsequent methods are very dependent on that. As someone who didn't write the code I need to bounce back and forth between `Installer.__init__` and the implementation classes since they have lost all context of ordering.

This is why I favor separate functions, but then again I haven't looked at this as deeply as you have. I trust your judgement if you think this is better.

Reply:
> The thing that bothers me in this solution is that the methods manage their own state anyway. `compute_location` will assign to different things depending on in which class it resides, and subsequent methods are very dependent on that. As someone who didn't write the code I need to bounce back and forth between `Installer.__init__` and the implementation classes since they have lost all context of ordering.

Yes, I know...